src/relax/op/op.cc

```diff
@@ -450,7 +538,9 @@ TVM_REGISTER_GLOBAL("relax.op.memory.alloc_tensor").set_body_typed(MakeMemAllocT
 RELAY_REGISTER_OP("relax.memory.kill_storage")
     .set_num_inputs(1)
     .add_argument("storage", "Expr", "The storage to be killed.")
-    .set_attr<FInferStructInfo>("FInferStructInfo", ReturnVoidStructInfo);
+    .set_attr<FInferStructInfo>("FInferStructInfo", ReturnVoidStructInfo)
+    // deallocation also isn't considered a "visible effect" as far as purity is concerned
+    .set_attr<Bool>("FPurity", Bool(false));
```

Comment:
the comment says that deallocation is not considered a visible effect, yet it is marked as impure?